examples/platform/silabs/provision/ProvisionStorageDefault.cpp

```diff
@@ -659,9 +659,44 @@ CHIP_ERROR Storage::GetProvisionRequest(bool & value)
 #if OTA_ENCRYPTION_ENABLE
 CHIP_ERROR Storage::SetOtaTlvEncryptionKey(const ByteSpan & value)
 {
+#if defined(SL_MBEDTLS_USE_TINYCRYPT)
+    // Tinycrypt doesn't support the key ID, so we need to store the key as a binary blob
+    return SilabsConfig::WriteConfigValueBin(SilabsConfig::kOtaTlvEncryption_KeyId, value.data(), value.size());
+#else  // MBEDTLS_USE_PSA_CRYPTO
     chip::DeviceLayer::Silabs::OtaTlvEncryptionKey::OtaTlvEncryptionKey key;
     ReturnErrorOnFailure(key.Import(value.data(), value.size()));
     return SilabsConfig::WriteConfigValue(SilabsConfig::kOtaTlvEncryption_KeyId, key.GetId());
+#endif // SL_MBEDTLS_USE_TINYCRYPT
+}
+
+CHIP_ERROR Storage::GetOtaTlvEncryptionKeyId(uint32_t & keyId)
+{
+#if defined(SL_MBEDTLS_USE_TINYCRYPT)
+    // Tinycrypt doesn't support the key ID
+    return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+#else  // MBEDTLS_USE_PSA_CRYPTO
+    // Read the key ID from the config
+    return SilabsConfig::ReadConfigValue(SilabsConfig::kOtaTlvEncryption_KeyId, keyId);
+#endif // SL_MBEDTLS_USE_TINYCRYPT
+}
+
+CHIP_ERROR Storage::DecryptUsingOtaTlvEncryptionKey(MutableByteSpan & block, uint32_t & ivOffset)
```

Comment:
I can't say I like this, but I don't know what can be done with Tinycrypt/mbedtls about this key... 
@rcasallas-silabs Any recommendation you could provide?

Reply:
hey @jmartinez-silabs , I agree with you, even @rcasallas-silabs agrees, but for time being we came to this conclusion. @rcasallas-silabs hopefully this is aligned with the discussion we had.

Reply:
I had a chat with @mkardous-silabs The usage of src/crypto/SessionKeystore.h and the implemented
src/crypto/RawKeySessionKeystore.h and/or src/crypto/PSASessionKeystore.h would clean up a lot of the added functions/ifdef in the code, and handle the PSA KeyIDs vs mbedtls/tinycrypt raw binary keys automatically with most of it being completely abstracted.

I can't spend the time to dig into this. But if you and Ricardo can look into that that would be great and the best solution.